File: SDKs/amazonaws.com/redshift/2012-12-01/python/src/sdk/models/shared/endpoint.py
import dataclasses
from typing import Optional
from ..shared import vpcendpointslist as shared_vpcendpointslist


@dataclasses.dataclass
class Endpoint:
    r"""Endpoint
    Describes a connection endpoint.
    """
    
    address: Optional[str] = dataclasses.field(default=None)
    port: Optional[int] = dataclasses.field(default=None)
    vpc_endpoints: Optional[list[shared_vpcendpointslist.VpcEndpointsList]] = dataclasses.field(default=None)
    

File: SDKs/amazonaws.com/redshift/2012-12-01/typescript/src/sdk/models/shared/endpoint.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { VpcEndpointsList } from "./vpcendpointslist";



// Endpoint
/** 
 * Describes a connection endpoint.
**/
export class Endpoint extends SpeakeasyBase {
  @SpeakeasyMetadata()
  address?: string;

  @SpeakeasyMetadata()
  port?: number;

  @SpeakeasyMetadata({ elemType: VpcEndpointsList })
  vpcEndpoints?: VpcEndpointsList[];
}
